qimage2ndarray/qimageview_python.py

```diff
@@ -15,11 +15,15 @@ def _re_buffer_address_match(buf_repr):
     return _re_buffer_address_match(buf_repr)
 
 def PySide_data(image):
-    # PySide's QImage.bits() returns a buffer object like this:
+    # PySide's QImage.bits() returns a memoryview object that can be used
+    # directly or a buffer object like this:
     # <read-write buffer ptr 0x7fc3f4821600, size 76800 at 0x111269570>
-    ma = _re_buffer_address_match(repr(image.bits()))
-    assert ma, 'could not parse address from %r' % (image.bits(), )
-    return (int(ma.group(1), 16), False)
+    bits = image.bits()
+    if "buffer" in repr(bits):
+        ma = _re_buffer_address_match(repr(image.bits()))
+        assert ma, 'could not parse address from %r' % (image.bits(), )
+        return (int(ma.group(1), 16), False)
+    return bits
 
 getdata = dict(
```

Comment:
Yes, that sounds reasonable.  I was thinking whether there was a more pythonic, duck-typing manner.  Maybe it would make sense to just instantiate a small QImage and inspect its `bits()` return value? That could automatically make it work across different versions of Python *and* PySide (as long as something supported is returned).  Anyhow, using the major py version would already be a big step forward (i.e. fix the most obvious case).